qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -1860,6 +1859,17 @@ def __json_encode__(self):
 
         # Handle non-serializable objects
         json_value["_jobs"] = self._safe_serialize_jobs()
+        service_value = getattr(self, "_service")
+        if service_value:
+            # the attribute self._service in charge of the connection and communication with the
+            #  experiment db. It doesn't have meaning in the json format so there is no need to serialize
+            #  it.
+            LOG.warning(
+                "Experiment was saved using %s service which cannot be automatically loaded. "
+                "Please reset service attribute if required",
+                str(type(service_value)),
+            )
+            json_value["_service"] = None
```

Comment:
I think this block could be re-written something like:

```python
# Set non-serializable attributes to None
for att in ["_service", "_backend"]:
    json_value[attr] = None
    value = getattr(self, attr):
    if value is not None:
        LOG.info("%s cannot be JSON serialized", type(value))
```

Reply:
changed the code to be as suggested below:
```python
for attr in ["_service", "_backend"]:
	json_value[attr] = None
	value = getattr(self, attr)
	if value is not None:
		LOG.info("%s cannot be JSON serialized", str(type(value)))
```
Resolving conversation.